service/awsconfig/v9/resource/ebsvolume/current.go

```diff
@@ -55,11 +55,27 @@ func (r *Resource) persistentVolumes(customObject v1alpha1.AWSConfig) (*EBSVolum
 	for _, vol := range output.Volumes {
 		// Volume is only included if it has a PV name tag.
 		if containsPersistentVolumeTag(vol.Tags) {
-			volumeIDs = append(volumeIDs, *vol.VolumeId)
+			attachments := []VolumeAttachment{}
+
+			if len(vol.Attachments) > 0 {
+				for _, a := range vol.Attachments {
+					attachments = append(attachments, VolumeAttachment{
+						Device:     *a.Device,
+						InstanceID: *a.InstanceId,
+					})
+				}
+			}
```

Comment:
If the volume has any attachments they are added to the state. So they can be detached prior to deletion.